components/style/rule_tree/mod.rs

```diff
@@ -580,13 +586,13 @@ impl StrongRuleNode {
                 let existing =
                     next_sibling_ptr.compare_and_swap(ptr::null_mut(),
                                                       new_ptr,
-                                                      Ordering::SeqCst);
+                                                      Ordering::Acquire);
```

Comment:
Yeah, so there are two different things that this controls:

 * Visibility of non-atomic/relaxed writes, and.
 * Compiler instruction re-ordering.

I'm still not seeing how some re-ordering gone wrong:

> So in this case, I believe we need Release when CAS-ing in the new entries, because otherwise the initialization of the new entry may not be "flushed" before the memory is accessed by other threads.

In that case, the relaxed CAS itself wouldn't be visible either, so there's no way any other thread would be able to access that pointer.